Userland/Libraries/LibGPU/Shader.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022, Stephan Unverwerth <s.unverwerth@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/RefCounted.h>
+
+namespace GPU {
+
+class Shader : public RefCounted<Shader> {
+public:
+    Shader(void const* ownership_token)
+        : m_ownership_token { ownership_token }
+    {
+    }
+
+    virtual ~Shader() = default;
+
+    void const* ownership_token() const { return m_ownership_token; }
+    bool has_same_ownership_token(Shader const& other) const { return other.ownership_token() == ownership_token(); }
+
+private:
+    void const* const m_ownership_token { nullptr };
```

Comment:
Why do we need this ownership token? Can't we refer back to `GPU::Device` instead?

Reply:
I just tried doing this with a `NonnullRefPtr<Device>` but that requires a lot of changes to the code. I'd rather leave this to another PR.

Reply:
Sure!